packages/babel-plugin-proposal-private-methods/test/fixtures/call-context/ts-as-class-field/output.js

```diff
@@ -0,0 +1,17 @@
+function _classPrivateFieldGet(receiver, privateMap) { var descriptor = privateMap.get(receiver); if (!descriptor) { throw new TypeError("attempted to get private field on non-instance"); } if (descriptor.get) { return descriptor.get.call(receiver); } return descriptor.value; }
+
+class A {
+  constructor() {
+    _x.set(this, {
+      writable: true,
+      value: void 0
+    });
+  }
+
+  fn() {
+    _classPrivateFieldGet(this, _x)();
```

Comment:
This still call still doesn't have the correct `this`!

Reply:
Thanks, admittedly I didn't look close enough and thought it was right! That's a tough one, because we're going in the opposite direction. The transformation is done by `babel-helper-member-expressions-to-functions`, which takes a member, and then checks to see if its parent is a call expression.

I was able to fix it with the following diff: https://hastebin.com/onotaqusiw.diff

It seems like this would be an issue for the entire helper. Maybe the helper should be a type stripper, instead of just something for calls? It would export a function for each direction.

The final tricky bit would be things like this:

`if (parentPath.isAssignmentExpression({ left: node })) {`

We could probably change that to this?

`if (parentPath.isAssignmentExpression({ left: previousParent })) {`

With previousParent set to node initially, and then the last parent each time we go a level up?

Reply:
It crossed my mind that if we move the logic to a plugin, we can't keep track of the previous parent. Maybe this logic has to be duplicated in this case.

Reply:
Maybe we could keep this for a follow-up PR then

Reply:
Sounds good. I have a first draft, so I'll open a WIP MR with that once this is merged.